vscode-dotnet-runtime-library/src/Utils/ExtensionConfigurationWorker.ts

```diff
@@ -2,21 +2,28 @@
 *  Licensed to the .NET Foundation under one or more agreements.
 *  The .NET Foundation licenses this file to you under the MIT license.
 *--------------------------------------------------------------------------------------------*/
-import { IExistingPath, IExtensionConfiguration } from '../IExtensionContext';
+import { IExistingPaths, IExtensionConfiguration, ILocalExistingPath } from '../IExtensionContext';
 import { IExtensionConfigurationWorker } from './IExtensionConfigurationWorker';
 
 export class ExtensionConfigurationWorker implements IExtensionConfigurationWorker {
     constructor(private readonly extensionConfiguration: IExtensionConfiguration,
                 private readonly pathConfigValueName: string | undefined) {}
 
-    public getPathConfigurationValue(): IExistingPath[] | undefined {
+    public getPathConfigurationValue(): IExistingPaths | undefined {
         return this.pathConfigValueName ? this.extensionConfiguration.get(this.pathConfigValueName) : undefined;
     }
 
-    public async setPathConfigurationValue(configValue: IExistingPath[]): Promise<void> {
+    public async setLocalPathConfigurationValue(configValue: ILocalExistingPath[]): Promise<void> {
```

Comment:
This code looks great, but I am concerned about whether VS Code will fail or not. The configuration is updated using the same key whether its local or global, so if the settings change b2b, then we could set the configuration to a string type, and later try to put a ILocalExistingPath[] obj in it. I'm not sure if VS Code will fail or how it will handle that, so we should either test it out, look at the vs code code, or use a dseparate configValueName for both global and local path

Reply:
Thanks for the explanation. The `setGlobalPathConfigurationValue` is got rid of for now.